tests/core/go_binary/pgo_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pgo_test
+
+import (
+	_ "embed"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/bazelbuild/rules_go/go/tools/bazel_testing"
+)
+
+//go:embed pgo.pprof
+var pgoProfile []byte
+
+func TestMain(m *testing.M) {
+	bazel_testing.TestMain(m, bazel_testing.Args{
+		Main: `
+-- src/BUILD.bazel --
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_test")
+
+go_binary(
+    name = "pgo_with_profile",
+    srcs = ["pgo.go"],
+    pgoprofile = ":pgo.pprof",
+)
+
+go_binary(
+    name = "pgo_without_profile",
+    srcs = ["pgo.go"],
+)
+
+go_test(
+    name = "compare_results_test",
+    data = [
+        ":pgo_with_profile",
+        ":pgo_without_profile",
+    ],
+    srcs = ["compare_results_test.go"],
+    deps = ["@io_bazel_rules_go//go/tools/bazel:go_default_library"],
+)
+
+-- src/pgo.go --
+package main
+
+import "fmt"
+
+func main() {
+  fmt.Println("Did you know that profile guided optimization was added to the go compiler in go version 1.20?")
+}
+
+-- src/compare_results_test.go --
+package compare_results_test
+
+import (
+    "crypto/sha256"
+    "os"
+    "strings"
+    "testing"
+
+    "github.com/bazelbuild/rules_go/go/tools/bazel"
+)
+
+func TestResultsAreSame(t *testing.T) {
+    rfs, err := bazel.ListRunfiles()
```

Comment:
This runfiles package is deprecated and I would prefer not adding new tests relying on it. Please use `.../go/runfiles` instead and either pass in the paths via `$(rlocationpath ...)` in `env` or look for `.exe` on Windows.

Reply:
This isn't using the [runfiles](https://pkg.go.dev/github.com/bazelbuild/rules_go@v0.41.0/go/runfiles) package and I didn't see ListRunfiles is deprecated in [github.com/bazelbuild/rules_go/go/tools/bazel](https://pkg.go.dev/github.com/bazelbuild/rules_go@v0.41.0/go/tools/bazel#ListRunfiles).

In any case, I don't understand how to apply your comment. If you would offer a full suggestion then I would be happy to accept it.

Reply:
I pushed a commit that relies on `cquery --output=files` in the `go_bazel_test` itself to get the paths, which I now think is more straightforward than looking up runfiles in the test. Please take a look.

Reply:
Thanks, that looks good to me!